Packages/com.unity.inputsystem/InputSystem/Plugins/iOS/iOSStepCounter.cs

```diff
@@ -0,0 +1,145 @@
+        public FourCC format => kFormat;
+
+        [InputControl(name = "stepCounter")]
+        public int stepCounter;
+    }
+
+    /// <summary>
+    /// Step Counter (also known as pedometer) sensor for iOS.
+    /// Note: You need to add 'Privacy - Motion Usage Description' to Info.plist to use this sensor.
+    /// </summary>
+    [InputControlLayout(stateType = typeof(iOSStepCounterState), variants = "StepCounter", hideInUI = true)]
+    public class iOSStepCounter : StepCounter
+    {
+        private const int kCommandFailure = -1;
+        private const int kCommandSuccess = 1;
+
+        internal delegate void OnDataReceivedDelegate(int deviceId, int numberOfSteps);
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct iOSStepCounterCallbacks
+        {
+            internal OnDataReceivedDelegate onData;
+        }
+
+        [DllImport("__Internal")]
+        private static extern int _iOSStepCounterEnable(int deviceId, ref iOSStepCounterCallbacks callbacks, int sizeOfCallbacks);
+
+        [DllImport("__Internal")]
+        private static extern int _iOSStepCounterDisable(int deviceId);
+
+        [DllImport("__Internal")]
+        private static extern int _iOSStepCounterIsEnabled(int deviceId);
+
+        [DllImport("__Internal")]
+        private static extern int _iOSStepCounterIsAvailable();
+
+        [DllImport("__Internal")]
+        private static extern int _iOSStepCounterIsAuthorized();
+
+        [MonoPInvokeCallback(typeof(OnDataReceivedDelegate))]
+        private static void OnDataReceived(int deviceId, int numberOfSteps)
+        {
+            var stepCounter = (iOSStepCounter)InputSystem.GetDeviceById(deviceId);
+            InputSystem.QueueStateEvent(stepCounter, new iOSStepCounterState {stepCounter = numberOfSteps});
+        }
+
+#if UNITY_EDITOR
+        private bool m_Enabled;
+#endif
+        public override unsafe long ExecuteCommand<TCommand>(ref TCommand command)
+        {
+            var ptr = UnsafeUtility.AddressOf(ref command);
+            var t = command.typeStatic;
+            if (t == QueryEnabledStateCommand.Type)
+            {
+#if UNITY_EDITOR
+                ((QueryEnabledStateCommand*)ptr)->isEnabled = m_Enabled;
+#else
+                ((QueryEnabledStateCommand*)ptr)->isEnabled = _iOSStepCounterIsEnabled(deviceId) != 0;
+#endif
+                return kCommandSuccess;
+            }
+
+            if (t == EnableDeviceCommand.Type)
+            {
+#if UNITY_EDITOR
+                m_Enabled = true;
+                return kCommandSuccess;
+#else
+                var callbacks = new iOSStepCounterCallbacks();
+                callbacks.onData = OnDataReceived;
+                return _iOSStepCounterEnable(deviceId, ref callbacks, Marshal.SizeOf(callbacks));
+#endif
+            }
+
+            if (t == DisableDeviceCommand.Type)
+            {
+#if UNITY_EDITOR
+                m_Enabled = true;
```

Comment:
should it be false here and on line 108 ?

Reply:
correct, nice find